db-parser prefix() support

bazsi opened the pull request:
This intends to fix #4130

@ryanfaircloth the original idea behind db-parser() was to be the "global" parser for any kind of data. For that reason, I kind of expected all name-value pairs to be specified right within the pattern database. It kind of failed in that mission, today I view db-parser() as a useful alternative to regexps, because it is MUCH faster. But in this role you are right, prefix() would make sense.

These are just a few preparatory patches, but I wanted to submit them so 1) you see there's progress, 2) I can get feedback via the automatic tests.

In the meanwhile, could you please describe your use-case how you'd apply db-parser() to it? It would be useful to learn from. Thanks.

ryanfaircloth commented:
I'm using it in a way that is close to the original idea. When I extract fields from the message I use a prefix so I can more easily map all of the extracted fields into a destination format ".fields.*" is great when I don't know what the fields will be but I need to map them to json in a destination format

bazsi commented:
This is now pretty much complete. Tests are running locally and now in GitHub CI.  Let's see if it turns green.

ryanfaircloth commented:
Looks green

ryanfaircloth commented:
@bazsi can this be merged?

bazsi commented:
removed WIP, rebased and added news file. I am trying to get some reviews so this can be merged.

In the meantime can you perhaps test this if it works for you? The CI produces a set of binaries, so you don't have to compile it yourself. Thanks

ryanfaircloth commented:
@bazsi sure how do I get bits from a PR?

bazsi commented:
The current run is here

https://github.com/syslog-ng/syslog-ng/actions/runs/3142427727

It is built by the "packages" github action for every PR. You will find the
packages among the artifacts, for every distro built.

On Wed, Sep 28, 2022, 14:18 Ryan Faircloth ***@***.***> wrote:

> @bazsi <https://github.com/bazsi> sure how do I get bits from a PR?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/syslog-ng/syslog-ng/pull/4133#issuecomment-1260819728>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFOK5RYS76L4HGYCSQGMJLWAQZRJANCNFSM6AAAAAAQMIYNJQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

ryanfaircloth commented:
Strange the objects are not downloadable must be a github issue I can try again in a bit

ryanfaircloth commented:
@bazsi I was able to download the images my current dev machine is now an apple with M1 I was unable to use the rpm the binary RPMs are x86_64

bazsi commented:
rebased against current master
